imapclient/acl_test.go

```diff
@@ -0,0 +1,100 @@
+package imapclient_test
+
+import (
+	"testing"
+
+	"github.com/emersion/go-imap/v2"
+)
+
+// TestACL runs tests on SetACL and MyRights commands (for now).
+func TestACL(t *testing.T) {
+	client, server := newClientServerPair(t, imap.ConnStateAuthenticated)
+
+	defer client.Close()
+	defer server.Close()
+
+	client.Create("MyFolder", nil)
+	client.Create("MyFolder/Child", nil)
+
+	for i, testcase := range []struct {
+		Mailbox   string
+		Rights    imap.RightSet
+		SetRights imap.RightSet
+	}{
+		// Testcase 1
+		// INBOX
+		{
+			Mailbox:   "INBOX",
+			Rights:    imap.AllRights,
+			SetRights: imap.AllRights,
+		},
+		// Testcase 2
+		// Custom folder
+		{
+			Mailbox:   "MyFolder",
+			Rights:    "rwi",
+			SetRights: "rwi",
+		},
+		// Testcase 3
+		// Custom child folder
+		{
+			Mailbox:   "MyFolder/Child",
+			Rights:    "rwi",
+			SetRights: "rwi",
+		},
+		// Testcase 4
+		// Add rights
+		{
+			Mailbox:   "MyFolder",
+			Rights:    "rwidc",
+			SetRights: "+dc",
+		},
+		// Testcase 5
+		// Remove rights
+		{
+			Mailbox:   "MyFolder",
+			Rights:    "rwi",
+			SetRights: "-dc",
+		},
+		// Testcase 6
+		// Set empty rights
+		{
+			Mailbox:   "MyFolder/Child",
+			Rights:    "",
+			SetRights: "",
+		},
+	} {
+		// execute SETACL command
+		err := client.SetACL(testcase.Mailbox, testUsername, testcase.SetRights).Wait()
+		if err != nil {
+			t.Fatalf("[%v] SetACL().Wait() error: %v", i+1, err)
```

Comment:
Instead of these `[%v]` prefixes, it would be clearer to use named sub-tests with `t.Run`.

Reply:
I wasn't aware of it, thanks